### ISSUE TITLE
The plugin doesn't work for android? This app doesn't support app configuration

### ISSUE THREAD

**Issue by ghuang0207** [audioeng/cordova-plugin-AppConfig#1]
I am using ionic2 with your plugin. We are using airwatch in our enterprise.

We are trying to get the value from the key: EnrollmentUser from user's device.

We published our code through Google Play, set as internal use only. We saw an error message in the airwatch console "This app doesn't support app configuration", when we tried to enable "Send Application Configuration" from the airwatch console. ( the same exact code works for iOS).

**Comment by ujjwalgarad** [comment]
I ran into the same exact issue and have been looking for the solution for many days.
It works of iOS but not for Android. This is very strange.
Did anyone find the solution yet?

**Comment by theaccordance** [comment]
Having the same problem as well.  Has anyone checked `platforms/android/AndroidManifest.xml` to see if the plugin injects the necessary properties?

**Comment by MrTechGadget** [comment]
That was my thought as well. This same problem has been logged with the
original oracle project that I forked this from.

**Comment by theaccordance** [comment]
If I'm reading the [Android documentation](https://developer.android.com/work/managed-configurations.html) correctly for configuring an application for managed configurations, the plugin needs to do the following to enable the functionality:
1.  Add a property to `platforms/android/AndroidManifest.xml` within the `<application>` block:

```<meta-data android:name="android.content.APP_RESTRICTIONS" android:resource="@xml/app_restrictions" />```

2.  Add a XML file which defines the properties exposed via AppConfig: `platforms/android/res/xml/app_restrictions.xml`

As far as I can tell looking at the plugin, it's doing neither.  My suspicion is supported by the fact that we don't define anything in Cordova's `config.xml` file to even define the restrictions, which is something I have to do for other plugins like Deeplinks.

**Comment by MrTechGadget** [comment]
Thank you for finding this, I suspect you are correct. I see you also
posted this on the oracle project page, thanks for that.

Now that they have fixed the issue with ionic in v1.0.1+, I plan on moving
my sample project back to using their plugin, so that is definitely the
best place to have that discussion.

**Comment by ghuang0207** [comment]
Hi Joshua,

Could you please let us know where is the plugin you were mentioning about?  I can try to build an ionic2 version sample code for this..  

Thank you to Joshua and Joe and Ujjwal :-)

George

"Now that they have fixed the issue with ionic in v1.0.1+, I plan on moving
my sample project back to using their plugin"

**Comment by lucaverrascina** [comment]
Maybe he was referring to cordova-plugin-emm-app-config by Oracle. I haven't tested it yet

**Comment by MrTechGadget** [comment]
Yes, I am referring to the original plugin by oracle that I forked. They have since fixed the problem that I forked it for, and there has been discussion about the manual steps a user must take to get it working on Android. 

I have a sample Ionic 2+ example project, you don’t need to write one yourself. https://github.com/audioeng/ionic-appconfig

I have done a little work merging it back to use the oracle plugin, but have not merged that back into master yet.
